mixins/visibility-mixin.js

```diff
@@ -0,0 +1,207 @@
+			},
+			final: {
+				opacity: '0',
+				transform: `translateY(-${moveYValue}px)`
+			},
+			finalDummy: {
+				height: '0'
+			},
+			thisOnTransitionStart: thisOnTransitionStart,
+			dummyOnTransitionEnd: dummyOnTransitionEnd
+		};
+		this._animateVisibility(animateHideRemoveStyle);
+	}
+
+	_animateRemove() {
+		const thisOnTransitionStart = () => {
+			this.dispatchEvent(new CustomEvent(
+				'd2l-visibility-mixin-remove-start',
+				{ bubbles: true, composed: false }
+			));
+		};
+		const dummyOnTransitionEnd = () => {
+			this.dummy.replaceWith(this);
+			this.dispatchEvent(new CustomEvent(
+				'd2l-visibility-mixin-remove-end',
+				{ bubbles: true, composed: false }
+			));
+			this.remove();
+		};
+		this._animateHideRemove(thisOnTransitionStart, dummyOnTransitionEnd);
+	}
+
+	_animateShow() {
+		this.style.display = this.displayOriginal;
+		const dummyOnTransitionStart = () => {
+			this.dispatchEvent(new CustomEvent(
+				'd2l-visibility-mixin-show-start',
+				{ bubbles: true, composed: false }
+			));
+		};
+		const thisOnTransitionEnd = () => {
+			this.dummy.replaceWith(this);
+			// done visibility transition, element is in original, fully visible state, so return the original transition to this
+			this.style.transition = this.transitionOriginal;
+			this.dispatchEvent(new CustomEvent(
+				'd2l-visibility-mixin-show-end',
+				{ bubbles: true, composed: false }
+			));
+		};
+		const animateShowStyle = {
+			initial: {
+				transition: `all ${transitionDuration}ms ease ${transitionDuration / 3}ms`,
+				opacity: '0',
+				transform: `translateY(-${moveYValue}px)`
+			},
+			initialDummy: {
+				transition: `height ${transitionDuration}ms ease`,
+				height: '0'
+			},
+			final: {
+				opacity: `${this.opacityOriginal}`,
+				transform: `${this.transformOriginal}`
+			},
+			finalDummy: {
+				height: `calc(${this.scrollHeight}px + ${window.getComputedStyle(this).marginTop} + ${window.getComputedStyle(this).marginBottom})`
+			},
+			dummyOnTransitionStart: dummyOnTransitionStart,
+			thisOnTransitionEnd: thisOnTransitionEnd
+		};
+
+		this._animateVisibility(animateShowStyle);
+	}
+
+	async _animateVisibility(animateStyle) {
+		if (!reduceMotion) {
+			Object.assign(this.style, animateStyle.initial);
+			Object.assign(this.dummy.style, animateStyle.initialDummy);
+
+			// we are in the middle of an earlier transition
+			if (document.body.contains(this.dummy)) {
+				// preserve the current opacity & transform for when we switch directions of transition
+				this.style.opacity = window.getComputedStyle(this).opacity;
+				this.style.transform = window.getComputedStyle(this).transform;
+
+				// preserve the current dummy height for when we switch directions of transition
+				this.dummy.style.height = window.getComputedStyle(this.dummy).height;
+				this.dummy.replaceWith(this);
+			}
+			this.replaceWith(this.dummy);
```

Comment:
Be careful with `replaceWith`... [caniuse says](https://caniuse.com/mdn-api_childnode_replacewith) it was only added in Safari 14. You can probably accomplish the same thing by `this.parentNode.insertBefore(dummy, this); dummy.appendChild(this);` I think.

Reply:
Looks like this works the same without issues. Fixed.